internal/xpkg/lint.go

```diff
@@ -30,21 +31,32 @@ import (
 )
 
 const (
-	errNotExactlyOneMeta         = "not exactly one package meta type"
-	errNotMeta                   = "meta type is not a package"
-	errNotMetaProvider           = "package meta type is not Provider"
-	errNotMetaConfiguration      = "package meta type is not Configuration"
-	errNotCRD                    = "object is not a CRD"
-	errNotXRD                    = "object is not an XRD"
-	errNotComposition            = "object is not a Composition"
-	errBadConstraints            = "package version constraints are poorly formatted"
-	errCrossplaneIncompatibleFmt = "package is not compatible with Crossplane version (%s)"
+	errNotExactlyOneMeta                 = "not exactly one package meta type"
+	errNotMeta                           = "meta type is not a package"
+	errNotMetaProvider                   = "package meta type is not Provider"
+	errNotMetaConfiguration              = "package meta type is not Configuration"
+	errNotCRD                            = "object is not a CRD"
+	errNotXRD                            = "object is not an XRD"
+	errNotMutatingWebhookConfiguration   = "object is not a MutatingWebhookConfiguration"
+	errNotValidatingWebhookConfiguration = "object is not an ValidatingWebhookConfiguration"
+	errNotComposition                    = "object is not a Composition"
+	errBadConstraints                    = "package version constraints are poorly formatted"
+	errCrossplaneIncompatibleFmt         = "package is not compatible with Crossplane version (%s)"
 )
 
 // NewProviderLinter is a convenience function for creating a package linter for
 // providers.
 func NewProviderLinter() parser.Linter {
-	return parser.NewPackageLinter(parser.PackageLinterFns(OneMeta), parser.ObjectLinterFns(IsProvider, PackageValidSemver), parser.ObjectLinterFns(IsCRD))
+	return parser.NewPackageLinter(parser.PackageLinterFns(OneMeta), parser.ObjectLinterFns(IsProvider, PackageValidSemver),
+		parser.ObjectLinterFns(
+			parser.Or(
+				parser.Or(
```

Comment:
I used double `OR` so that we don't have to bump crossplane-runtime in a patch release just to get the new interface of `parser.Or` that can accept more than two arguments.